### PR TITLE
Fixes #15238 - ignore empty NIC identifiers

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by fwdIT** [comment]
http://pastebin.com/s6kx7VYk

**BEFORE PATCH**

2016-05-31 13:07:26 [app] [D] #### TFTP: def no_errors host.errors.empty? = true
2016-05-31 13:07:26 [app] [D] #### TFTP: def no_errors errors.empty? = **false**
2016-05-31 13:07:26 [app] [D] #### TFTP: errors: @messages={:identifier=>["has already been taken"]}

**AFTER PATCH**

2016-05-31 14:44:14 [app] [D] #### TFTP: def no_errors errors.empty? = true
2016-05-31 14:44:14 [app] [D] #### TFTP: def no_errors host.errors.empty? = true
2016-05-31 14:44:14 [app] [D] #### TFTP: def generate_pxe_template

-> TFTP menu now gets rewritten after successful build due to no more errors

**Comment by ares** [comment]
[test] PuppetclassLookupKeyJSTest.test_0002_uncheck override only on pg/2.0

**Comment by brandonweeks** [comment]
👍

**Comment by ares** [comment]
[test]

**Comment by dLobatog** [comment]
Test passes even without the changes - so it's not really testing this behavior. Otherwise :+1: !

File: test/unit/nics/base_test.rb [hunk 1]
```diff
@@ -106,10 +106,25 @@ class NicBaseTest < ActiveSupport::TestCase
...
         refute nic.valid?
         assert nic.errors.has_key?(:identifier)
       end
+
+      test 'it ignores empty identifiers' do
+        nic.mac = '00:11:11:22:22:33'
+        nic2.mac = '00:11:11:22:22:34'
+        nic.identifier = nic2.identifier = ''
+        nic.save
+        nic2.save
```
Comment (dLobatog) [inline test/unit/nics/base_test.rb]: nitpick - nic2.save doesn't need to be called here, just `.valid?` will fail without the patch as there exists a `nic` with identifier `' '`. I'll remove it on merge if jenkins is :green_apple:

**Comment by dLobatog** [comment]
Merged as 7e20cbac5951b858716268cce29f0fad16b71081, thanks @ares!

**Pull request closed without merging**